PopPUNK/__main__.py

```diff
@@ -280,31 +287,54 @@ def main():
 
         # generate sketches and QC sequences
         createDatabaseDir(args.output, kmers)
-        seq_names = constructDatabase(
-                        args.r_files,
-                        kmers,
-                        sketch_sizes,
-                        args.output,
-                        args.threads,
-                        args.overwrite,
-                        codon_phased = args.codon_phased,
-                        calc_random = True)
-
-        rNames = seq_names
-        qNames = seq_names
-        refList, queryList, distMat = queryDatabase(rNames = rNames,
-                                                    qNames = qNames,
-                                                    dbPrefix = args.output,
-                                                    queryPrefix = args.output,
-                                                    klist = kmers,
-                                                    self = True,
-                                                    number_plot_fits = args.plot_fit,
-                                                    threads = args.threads)
-        qcDistMat(distMat, refList, queryList, args.max_a_dist)
-
-        # Save results
-        dists_out = args.output + "/" + os.path.basename(args.output) + ".dists"
-        storePickle(refList, queryList, True, distMat, dists_out)
+        seq_names_passing = \
+            constructDatabase(
+                args.r_files,
+                kmers,
+                sketch_sizes,
+                args.output,
+                args.threads,
+                args.overwrite,
+                codon_phased = args.codon_phased,
+                calc_random = True)
+
+        distMat = queryDatabase(rNames = seq_names_passing,
+                                qNames = seq_names_passing,
+                                dbPrefix = args.output,
+                                queryPrefix = args.output,
+                                klist = kmers,
+                                self = True,
+                                number_plot_fits = args.plot_fit,
+                                threads = args.threads)
+        names_to_remove = qcDistMat(distMat,
+                                seq_names_passing,
+                                seq_names_passing,
+                                args.max_pi_dist,
+                                args.max_a_dist,
+                                args.reference_isolate)
+        
+        # prune based on distance from reference if provided
+        if args.reference_isolate is not None and len(names_to_remove) > 0 and args.qc_filter == "prune":
```

Comment:
Can do - could do it semi-systematically based on sequence statistics stored in database (e.g. minimial missing sequence)?

Reply:
I suppose sometimes a user might have a genuine reference in the database, but that's probably unusual. I think having a `--prune-distance` flag which just picks one automatically (as you say, lowest Ns near the mean length would make sense) would be fine

Reply:
I think it's worth keeping this functionality, as it's likely users will sometimes want to specify a reference - either because there is a lot of contamination, or as a way of subseting a complex collection of sequences (simple example is something like pulling the Shigella sequences out of a Shigella/E. coli mix).

Reply:
Changed in a87c94d.

Reply:
Done in 40a6e4f.